### PR TITLE
Maintenance fixes

### LINKED ISSUE #190: Cleanup the namespace and the arguments
While reviewing the module, I noticed that there isn't a clean namespace. In prep for adding pipelines to the module, we should restructure it. 
- We should have the read and write functions live in `mne_bids.io`. this would be either with the function signature there, or having the name called from the io `__init__` file. 
- ~~We have `make_bids_basename` but it does not include the bids_directory base. I think we should include it in the function signature and have the full basename path.~~
- ~~Afterwards we could then remove bids_root from the `read_raw_bids` given that the basename would now contain this information~~

### PR DESCRIPTION
**Pull request by sappelhoff**
several minor fixes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,9 +17,9 @@ Installation
 
 We recommend the [Anaconda](https://www.anaconda.com/download/) Python
 distribution. We require that you use Python 3.
-Besides `numpy`, `scipy` and `matplotlib` (which are included in the standard
-Anaconda installation), you will need to install the most recent version of MNE
-using the `pip` tool:
+Besides `numpy` and `scipy` (which are included in the standard Anaconda
```
Comment (jasmainak) [inline README.md]: shouldn't we also update the CI if we do this? otherwise we won't know ...

File: examples/convert_group_studies.py [hunk 2]
```diff
@@ -69,8 +69,9 @@
                             'run_%02d_raw.fif' % run)
 
         raw = mne.io.read_raw_fif(raw_fname)
-        bids_basename = ('sub-%02d_ses-01_task-VisualFaces_run-%d'
-                         % (subject_id, run))
+        bids_basename = make_bids_basename(subject=str(subject_id),
+                                           session='01', task='VisualFaces',
+                                           run=str(run))
```
Comment (jasmainak) [inline examples/convert_group_studies.py]: why `str(..)` ?

File: README.md (hunk 1 above)
Comment (sappelhoff) [reply README.md]: CIs are already updated. When I did it, I just missed to remove Matplotlib from the README

File: examples/convert_group_studies.py (hunk 2 above)
Comment (sappelhoff) [reply examples/convert_group_studies.py]: because make_bids_basename expects a string

File: mne_bids/io.py [hunk 3]
```diff
@@ -105,9 +105,17 @@ def read_raw_bids(bids_fname, bids_root, return_events=True,
...
-        Weather to return events or not. Default is True.
+        Whether to return events or not. Default is True.
     verbose : bool
         The verbosity level
+
+    Returns
+    -------
+    raw : instance of Raw
+        The data as MNE-Python Raw object.
+    events : ndarray
```
Comment (jasmainak) [inline mne_bids/io.py]: you are missing a description here

File: mne_bids/io.py [hunk 4]
```diff
@@ -105,9 +105,17 @@ def read_raw_bids(bids_fname, bids_root, return_events=True,
...
+    event_id : dict
```
Comment (jasmainak) [inline mne_bids/io.py]: same here

File: examples/rename_brainvision_files.py [hunk 5]
```diff
@@ -78,4 +78,4 @@
 #
 # Lastly, there is a tool to check the integrity of your BrainVision files.
 # For that, see the
-# `BrainVision Validator <https://github.com/sappelhoff/brainvision-validator>`_
+# `BrainVision Validator <https://github.com/sappelhoff/brainvision-validator>`
```
Comment (jasmainak) [inline examples/rename_brainvision_files.py]: are you sure this works after removing the underscore?

**Comment by jasmainak** [comment]
why are the CIs failing?

File: examples/rename_brainvision_files.py (hunk 5 above)
Comment (sappelhoff) [reply examples/rename_brainvision_files.py]: no, thanks for the catch ... I thought it was trailing whitespace O.o

**Comment by sappelhoff** [comment]
Travis says: `This job was cancelled because the "Auto Cancellation" feature is currently enabled, and a more recent build (#865) for pull request #191 came in while this job was waiting to be processed.`

But tests pass locally and I think we can also trust AppVeyor. When CircleCI is green, this is ready to be merged.

**Comment by jasmainak** [comment]
Thanks @sappelhoff !

**Pull request merged**